### ISSUE TITLE
Not downloading Daily reports from MacAppStore

### ISSUE THREAD

**Issue by Lymes** [omz/AppSales-Mobile#279]
ReportDownloadOperation.m,
 -(void)main
```
NSData *reportData = [NSURLConnection sendSynchronousRequest:reportDownloadRequest returningResponse:&response error:NULL];
NSString *errorMessage = [[response allHeaderFields] objectForKey:@"Errormsg"];
if ( errorMessage )
{
        NSLog( @"  %@", errorMessage );
}
else if ( reportData )
```
After 5 November 2015 always prints the following error message:

`AppSalesMobile[29627:925630]   Daily reports are only available for past 365 days. Please enter a new date.`

Before that date the app worked fine.  I've checked out the query params
```
NSString *reportDownloadBodyString = [NSString stringWithFormat:@"USERNAME=%@&PASSWORD=%@&VNDNUMBER=%@&TYPEOFREPORT=%@&DATETYPE=%@&REPORTTYPE=%@&REPORTDATE=%@",
                                                  escapedUsername, escapedPassword, vendorID, @"Sales", dateType, @"Summary", reportDateString]; 
```
and the params seem to be correct.

**Comment by ddaddy** [comment]
Reports since 5th November are not available. See iTC reports section for the notice.

Reports for 5th were actually released but then removed. So if you have a 5th report you should delete it until Apple release a corrected one.

**Comment by Lymes** [comment]
Oh man, thank you.  BTW, I've forked your repository, great job!

**Comment by ddaddy** [comment]
👍🏼
Apple also say reports between 26th Oct - 2nd Nov are also wrong and will eventually be reissued.

**Comment by Morpheus2002** [comment]
@ddaddy stupid question: how do I redownload old reports in ITC? I don't see any option for that. Now, that reports between 26th Oct - 2nd Nov are corrected I would like to replace the wrong ones.

**Comment by ddaddy** [comment]
You just delete the ones you have. Press and hold on a bar to delete.

**Comment by dave256** [comment]
I think we'll also need to edit the file: ReportDownloadOperation.m and edit the line:
NSInteger maxNumberOfAvailableReports = [dateType isEqualToString:@"Daily"] ? 14 : 13;

You'll want to change the 14 to about 28-30 to make it look for the earlier dates. I haven't done this yet, but plan to try it later today.

**Comment by ddaddy** [comment]
I'm not sure if that's different in my fork, but mine downloaded them no problem.

**Comment by dave256** [comment]
@ddaddy yes, it is set to 30 in your fork. As I suspected, it did not work with 14, but changing it to 30 allowed me to re-download the new reports after deleting them.

**Comment by Morpheus2002** [comment]
Thank you guys! It worked for AppSales. But still don't see the option for manual report download in ITC.

**Comment by ddaddy** [comment]
In iTC it's slightly hidden.
Go to the Sales and Trends page, then press at the top left where it says Top Content and you can change it to reports.

**Comment by Morpheus2002** [comment]
@ddaddy you are the best! 😊
